x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/extractor/ProcessedField.java

```diff
@@ -52,7 +52,7 @@ public ProcessedField(PreProcessor processor) {
             }
         }
         preProcessor.process(inputs);
-        return preProcessor.outputFields().stream().map(inputs::get).toArray();
+        return preProcessor.outputFields().stream().map(inputs::get).filter(Objects::nonNull).toArray();
```

Comment:
Does this work correctly? If we filter out `null` objects, won't we mess the correspondence of the values to the output fields?

Reply:
Let me think on this more. 

We don't want to return partial lists, for sure. But we also don't want to put empty/missing unless the caller supports missing values...